src/components/header/Header.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import SearchBox from "../searchbox/SearchBox";
+import ProfileMenu from "../../pages/homepage/ProfileMenu";
+import { useNavigate } from "react-router-dom"; // Import useNavigate
+import { toast } from "sonner";
+import styled from "styled-components";
+
+export default function Header(props: { onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void }) {
```

Comment:
I recommend using arrow functions. It's not a must though.

On the other hand, defining `props` outside of the function definition would improve readability. For example, by using an interface or type such as:
```typescript
interface HeaderProps {
  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Header: React.FC<HeaderProps> = ({ onSearchChange }) => {
```

Or something like that

Reply:
ok, I adjusted it